File: Mobile/reunionou_app/lib/models/Event.dart
import 'package:reunionou_app/models/EventAuthor.dart';
import 'package:reunionou_app/models/Location.dart';
import 'package:reunionou_app/models/User.dart';

class Event{
  int id;
  EventAuthor author;
  String title;
  String description;
  String date;
  String token;
  Location location;//var location = {"id" : 1, "name" : "tour eiffel"......};
  bool isOwned;
  List<User> invitedUser = [];
  double distanceBetweenAuth;

  Event(this.id, this.author, this.title, this.description, this.date, this.token, this.location, this.isOwned);
}

File: API/src/ReunionouAPI/Models/Event.php
<?php
namespace ReunionouAPI\Models;

use Illuminate\Database\Eloquent\Model;

class Event extends Model
{
    
    protected $table = 'events';
    protected $primaryKey = 'id';
    public $timestamps = false;

    protected $fillable = ['title', 'description', 'date', 'location_id', 'user_id', 'token'];

    public function location(){
        return $this->belongsTo('ReunionouAPI\Models\Location', 'location_id');
    }

    public function author(){
        return $this->belongsTo('ReunionouAPI\Models\User', 'user_id');
    }

}